### PR TITLE
remove calls to projectm text helpers that are no longer available

### LINKED ISSUE #43: Doesn't build against projectM HEAD due to missing text-related helpers
https://github.com/projectM-visualizer/projectm/pull/652 removed parts of projectM this frontend uses.

The errors I get when building the frontend are

```
% make
Consolidate compiler generated dependencies of target projectMSDL
[ 11%] Building CXX object src/CMakeFiles/projectMSDL.dir/AudioCapture.cpp.o
[ 22%] Building CXX object src/CMakeFiles/projectMSDL.dir/ProjectMSDLApplication.cpp.o
[ 33%] Building CXX object src/CMakeFiles/projectMSDL.dir/ProjectMWrapper.cpp.o
[ 44%] Building CXX object src/CMakeFiles/projectMSDL.dir/RenderLoop.cpp.o
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:59:21: error: use of undeclared identifier 'projectm_is_text_input_active'
                if (projectm_is_text_input_active(_projectMHandle, true))
                    ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:70:22: error: use of undeclared identifier 'projectm_is_text_input_active'
                if (!projectm_is_text_input_active(_projectMHandle, true))
                     ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:85:21: error: use of undeclared identifier 'projectm_is_text_input_active'
                if (projectm_is_text_input_active(_projectMHandle, true))
                    ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:87:21: error: use of undeclared identifier 'projectm_set_search_text'; did you mean 'projectm_get_search_index'?
                    projectm_set_search_text(_projectMHandle, event.text.text);
                    ^~~~~~~~~~~~~~~~~~~~~~~~
                    projectm_get_search_index
/usr/local/include/libprojectM/projectM.h:822:30: note: 'projectm_get_search_index' declared here
PROJECTM_EXPORT unsigned int projectm_get_search_index(projectm_handle instance, const char* name);
                             ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:88:21: error: use of undeclared identifier 'projectm_populate_preset_menu'
                    projectm_populate_preset_menu(_projectMHandle);
                    ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:167:13: error: use of undeclared identifier 'projectm_set_toast_message'
            projectm_set_toast_message(_projectMHandle, aspectCorrectionEnabled ? "Aspect Correction Enabled"
            ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:196:17: error: use of undeclared identifier 'projectm_set_toast_message'
                projectm_set_toast_message(_projectMHandle, _audioCapture.AudioDeviceName().c_str());
                ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:234:13: error: use of undeclared identifier 'projectm_set_toast_message'
            projectm_set_toast_message(_projectMHandle, shuffleEnabled ? "Shuffle Enabled" : "Shuffle Disabled");
            ^
/Users/noah/src/frontend-sdl2/src/RenderLoop.cpp:340:13: error: use of undeclared identifier 'projectm_delete_search_text'
            projectm_delete_search_text(_projectMHandle);
            ^
9 errors generated.
make[2]: *** [src/CMakeFiles/projectMSDL.dir/RenderLoop.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/projectMSDL.dir/all] Error 2
make: *** [all] Error 2
```

Workaround is to build projectM from and older version:

```
cd projectm
git reset --hard 62cecb36d6c796771f84511347a5772b48162960
cmake && make && make install
```

### PR DESCRIPTION
**Pull request by nzoschke**
Replace calls to projectm text helpers removed in projectM-visualizer/projectm#652 with logging. Fixes #43

### CODE REVIEW

**Comment by nzoschke** [comment]
CC @kblaschke based on your comment on https://github.com/projectM-visualizer/projectm/pull/652

> Also note this commit will break the rewritten SDL2 frontend for now, but that's fine. I'll fix that after the API changes are finished.

If you have a branch of `frontend-sdl2` that addresses this I'm happy to review and work off that. 

But I think it would be nice to get master / master building again to base other frontend improvements on.

File: src/RenderLoop.cpp [hunk 1]
```diff
@@ -164,16 +146,15 @@ void RenderLoop::KeyEvent(const SDL_KeyboardEvent& event, bool down)
         {
             bool aspectCorrectionEnabled = !projectm_get_aspect_correction(_projectMHandle);
             projectm_set_aspect_correction(_projectMHandle, aspectCorrectionEnabled);
-            projectm_set_toast_message(_projectMHandle, aspectCorrectionEnabled ? "Aspect Correction Enabled"
-                                                                                : "Aspect Correction Disabled");
+            SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, aspectCorrectionEnabled ? "Aspect Correction Enabled" : "Aspect Correction Disabled");
```
Comment (nzoschke) [inline src/RenderLoop.cpp:149]: is there a line length enforced in the repo?

**Comment by kblaschke** [comment]
I've not yet started to refactor the frontend, besauce there are plenty more API changes to come, e.g. moving the playlist functionality into another library with different API functions.

We can keep this PR open and keep it in sync to the ongoing changes in the main repo though, this will probably help some people to build the application against the latest upstream master. When the API changes are done, simply merge the PR and that's it.

File: src/RenderLoop.cpp (hunk 1 above)
Comment (kblaschke) [inline src/RenderLoop.cpp:149]: Please use the POCO logging macros, otherwise the application will output stuff in two different places.

**Comment by nzoschke** [comment]
Ok That works for me. 

I'm able to maintain my own front end fork no problem for now. 

Just sharing some of the pieces in case it helps others.

**Review by kblaschke — CHANGES_REQUESTED** [review]

**Comment by kblaschke** [comment]
Opened another PR with a large number of other required changes, as libprojectM changed some more in the meantime.

Closing this PR, as the same change is also in the other one.

**Pull request closed without merging**